### ISSUE TITLE
Field's resolvers not working

### ISSUE THREAD

**Issue by maxpain** [join-monster/join-monster#74]
```javascript
const Match = new GraphQLObjectType({
	description: 'A match type',
	name: 'Match',
	sqlTable: 'matches',
	uniqueKey: 'id',
	fields: () => ({
		id: { type: new GraphQLNonNull(GraphQLID) },
		members: {
			type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(MatchMember))),
			// sqlJoin: (matchesTable, matchMembersTable) =>
			// 	`${matchMembersTable}.match_id = ${matchesTable}.id`

			where: (matchMembersTable, args, { matchID }) => `
				${matchMembersTable}.match_id = ${matchID}
			`,

			resolve(root, args, context, resolveInfo) {
				context.matchID = root.id

				return joinMonster(resolveInfo, context, sql =>
					postgres.query(sql)
				, { dialect: 'pg' })
			}
		}
      })
})
```

Query:
```graphql
{
  match {
    id
    members {
      id
    }
  }
}
```

Error:
```json
{
  "data": {
    "match": null
  },
  "errors": [
    {
      "message": "syntax error at or near \"FROM\"",
      "locations": [
        {
          "line": 8,
          "column": 3
        }
      ],
      "path": [
        "match"
      ]
    }
  ]
}
```

Logs:
```javascript
SQL_AST
  API  { type: 'table',
  API   name: 'matches',
  API   as: 'match',
  API   fieldName: 'match',
  API   grabMany: false,
  API   where: [Function: where],
  API   children:
  API    [ { type: 'column', name: 'id', fieldName: 'id', as: 'id' },
  API      { type: 'column', name: 'id', fieldName: 'id', as: 'id' },
  API      { type: 'table',
  API        name: 'match_members',
  API        as: 'members',
  API        fieldName: 'members',
  API        grabMany: true,
  API        where: [Function: where],
  API        children:
  API         [ { type: 'composite',
  API             name: [ 'match_id', 'user_id' ],
  API             fieldName: 'match_id#user_id',
  API             as: 'match_id#user_id' },
  API           { type: 'columnDeps',
  API             names: { match_id: 'match_id', user_id: 'user_id' } } ] },
  API      { type: 'columnDeps', names: {} } ] }
  API  +848ms
  API Thu, 16 Feb 2017 16:59:41 GMT join-monster
  API SQL
  API  SELECT
  API   "match"."id" AS "id",
  API   NULLIF(CONCAT("members"."match_id", "members"."user_id"), '') AS "members__match_id#user_id",
  API   "members"."match_id" AS "members__match_id",
  API   "members"."user_id" AS "members__user_id"
  API FROM matches AS "match"
  API FROM match_members AS "members"
  API WHERE "match".id = 9 AND
  API 				"members".match_id = undefined
  API
  API  +282ms
  API Thu, 16 Feb 2017 16:59:41 GMT join-monster
  API SHAPE_DEFINITION
  API  { id: 'id',
  API   members:
  API    [ { 'match_id#user_id': 'members__match_id#user_id',
  API        match_id: 'members__match_id',
  API        user_id: 'members__user_id' } ] }
  API  +1ms
  API error: syntax error at or near "FROM"
  API     at Connection.parseE (/Users/maxpain/dev/projects/fastcup/backend/node_modules/pg/lib/connection.js:554:11)
  API     at Connection.parseMessage (/Users/maxpain/dev/projects/fastcup/backend/node_modules/pg/lib/connection.js:381:17)
  API     at Socket.<anonymous> (/Users/maxpain/dev/projects/fastcup/backend/node_modules/pg/lib/connection.js:117:22)
  API     at emitOne (events.js:96:13)
  API     at Socket.emit (events.js:189:7)
  API     at readableAddChunk (_stream_readable.js:176:18)
  API     at Socket.Readable.push (_stream_readable.js:134:10)
  API     at TCP.onread (net.js:551:20)
  API  +58ms
```

**Comment by acarl005** [comment]
@Maxpain177 can i see the `MatchMember` type?

**Comment by maxpain** [comment]
@acarl005, Yes.

```javascript
export const MatchMember = new GraphQLObjectType({
	description: 'A match member type',
	name: 'MatchMember',
	sqlTable: 'match_members',
	uniqueKey: ['match_id', 'user_id'],
	fields: () => ({
		id: {
			type: new GraphQLNonNull(GraphQLID),
			sqlDeps: ['match_id', 'user_id'],
			resolve: m => `${m.match_id}_${m.user_id}`
		},
		side: { type: new GraphQLNonNull(MatchMemberSide) },
		ready: { type: new GraphQLNonNull(GraphQLBoolean) },
		user: {
			type: new GraphQLNonNull(User),
			sqlJoin: (matchMembersTable, usersTable) =>
				`${usersTable}.id = ${matchMembersTable}.user_id`
		},
		kills: { type: new GraphQLNonNull(GraphQLInt) },
		deaths: { type: new GraphQLNonNull(GraphQLInt) },
		assists: { type: new GraphQLNonNull(GraphQLInt) }
	})
})
```

**Comment by acarl005** [comment]
@Maxpain177 Your `MatchMember` type is mapped to a table since `sqlTable` is set to `'match_members'`. As it is a child of the `Match` type, which is also mapped to a table, Join Monster expects the `members` field to define either `sqlJoin`, `sqlJoins`, `sqlBatch`, to tell it how to join/batch the data.

This should throw a better error, rather than attempting to generate the SQL. I have published an update **v1.0.0-beta.5** that performs the validation and produces a more helpful error message.
